Client/mods/deathmatch/logic/luadefs/CLuaWorldDefs.cpp

```diff
@@ -58,6 +58,7 @@ void CLuaWorldDefs::LoadFunctions()
                                                                              {"getFPSLimit", GetFPSLimit},
                                                                              {"getBirdsEnabled", GetBirdsEnabled},
                                                                              {"getCoronaReflectionsEnabled", ArgumentParser<GetCoronaReflectionsEnabled>},
+                                                                             {"getWorldProperty", ArgumentParserWarn<false, GetWorldProperty>},
```

Comment:
Is this an entirely new Lua function/definition? If so, use `ArgumentParser`, not `ArgumentParserWarn`

Reply:
This is new Lua function. However, if I use `ArgumentParser` then it will throw an error on incorrect / missing property name, wouldn't it?
It's not a problem for me personally but other similar functions throw warnings right now. Also this function is likely to be used in a loop (may cause unwanted loop breaks if we add new properties in the future and scripter forget to add client version checks)

Reply:
All new functions should use `ArgumentParser` and are hard error on **usage** failure. Even when modifying existing functions, they should be switched to use `ArgumentParserWarn` if possible.

`ArgumentParserWarn` is just for compatibility so upgraded definitions produce a warning instead of error.

Currently you don't actually `throw` in the function, so you'll want to introduce a `default` case to the switch statement and throw in there, ie:

```cpp
default:
    throw std::invalid_argument{"Property not supported"};
```

You can also remove the `return false;` on the last line of the method, as it's not necessary by introducing this.

We shouldn't worry about user scripts failing because they used it wrong, they should worry about using it correctly ;)

Reply:
It seems that default error message produced by ArgumentParser is more than enough to understand what is wrong in this case. So no need to throw custom error message. Thank you for this information regarding new argument parser.